### ISSUE TITLE
`<figcaption>` not inheriting layout of figure in HTML.

### ISSUE THREAD

**Issue by batpigandme** [quarto-dev/quarto-cli#8666]
### Bug description

Currently, the `<figcaption>` of an image/figure with a layout class (e.g. `column-body-outset`) is not having that same class applied.
<img width="1429" alt="Image with outset alignment, figure below it has narrower alignment because the class is not applied." src="https://github.com/quarto-dev/quarto-cli/assets/831732/2366ea7b-e2a3-470b-b460-b7066c10407b">

### Steps to reproduce

Here's a GitHub repo with a reproducible example (the [index.qmd](https://github.com/batpigandme/quarto-figcap-demo/blob/main/index.qmd) file): <https://github.com/batpigandme/quarto-figcap-demo>.

The rendered site is <https://mara.quarto.pub/quarto-figcap-demo/>.

### Expected behavior

I would expect that the `<figcaption>` would have the same layout as the figure itself (i.e. what happens when I apply `.column-body-outset` to `<figcaption>`, as seen below.
<img width="1429" alt="Figure with caption aligned with image so both have column-body-outset. Annotation added to show that there is an Image with class  .column-body-outset, and, below .column-body-outset class manually applied to figcaption." src="https://github.com/quarto-dev/quarto-cli/assets/831732/bc233c09-e9a2-4c95-9c65-598ef904d9a8">

Barring that, I would expect the layout in the RStudio Visual Editor to somehow reflect that the caption is not inside of the parent div with the layout class applied (obviously this would be a different issue in that repo).
<img width="1036" alt="Image and caption inside of the column-body-outset as they appear in the RStudio visual editor." src="https://github.com/quarto-dev/quarto-cli/assets/831732/24dfa5fe-bc4d-474c-b003-7a7c80e9ad45">

### Actual behavior

`.column-body-outset` is applied only to the`<img>`, and not the `<figcaption>`.

### Your environment

- RStudio IDE: 2024.04.0-daily+320
-OS: Mac Ventura 13.6.3

### Quarto check output

```bash
Quarto 99.9.9
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 99.9.9
      Path: /Users/maraaverick/quarto-cli/package/dist/bin

Check file:///Users/maraaverick/quarto-cli/src/resources/vendor/deno-land/x/puppeteer@9-0-2/mod.ts
[✓] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/maraaverick/Library/TinyTeX/bin/universal-darwin
      Version: 2022

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.8.11 (Conda)
      Path: /opt/homebrew/Caskroom/miniconda/base/bin/python
      Jupyter: 5.3.0
      Kernels: julia-1.8, python3

(-) Checking Jupyter engine render....Unable to load extension: pydevd_plugins.extensions.types.pydevd_plugin_pandas_types
(|) Checking Jupyter engine render....Unable to load extension: pydevd_plugins.extensions.types.pydevd_plugin_pandas_types
[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.2.3
      Path: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources
      LibPaths:
        - /Users/maraaverick/Library/R/arm64/4.2/library
        - /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
      knitr: 1.45
      rmarkdown: 2.25.3

[✓] Checking Knitr engine render......OK
```

**Comment by batpigandme** [comment]
I just checked an older deploy of the same page on which I had this issue ([see here](https://65b665d4c072c90008e56fd6--quartoandme.netlify.app/css-grid-and-accessibility.html), also in screenshot below), and it appears that the entire figure was in a parent `<div.column-body-outset>`, which I actually think makes more sense in terms of the syntax being used.
<img width="1423" alt="Older version of document where parent div of image and figcaption share the same layout from column-body-outset." src="https://github.com/quarto-dev/quarto-cli/assets/831732/e11a0471-64f9-457a-b0f6-72faa89b1dc5">
Unfortunately, I didn't record what Quarto version that used (it could be well over a year ago).

**Comment by cscheid** [comment]
Hm. We did change the HTML intentionally. We document it here: https://quarto.org/docs/prerelease/1.4/crossref.html#changes-in-html-output

But I think what you're describing makes sense, and we just didn't realize the consequence.
We needed to change that in order for our DOM to support the different crossref use cases that are possible in 1.4.

It's not really a _regression_, but I think we need to improve this.

**Comment by batpigandme** [comment]
Thanks for the clarification and pointer to the docs, @cscheid (sorry I missed that before filing this).

I agree, it's definitely not a clear regression, per se. I played around with some more examples and added them to the GH repo and page I linked to in the issue.

There are definitely a couple instances where it looks “off” to me—namely when you have images near each other where the caption is clearly aligned to the left of one and not the other (see screenshots below)—but they're admittedly edge cases, and it's not _that_ off looking (it's obvious enough to the eye that the captions are basically aligned to the body text).

![Screenshot with column-page-left image above column-screen-right image.](https://github.com/quarto-dev/quarto-cli/assets/831732/0ca9ca84-8727-41ac-82f1-074f2410d7cb)
 
<img width="1618" alt="Screenshot with column-screen-right image above column-screen-left image." src="https://github.com/quarto-dev/quarto-cli/assets/831732/f5164643-4e73-4d8b-bffa-be746d202b4e">

**Comment by cscheid** [comment]
Thanks - these examples are super helpful for us.

**Comment by dragonstyle** [comment]
If we end up adapting the behavior, I think we'll have to make this configurable. Depending upon how figure captions are used, it will often make sense for the captions to follow the body column (for example, in an academic article with narrative and figures which themselves have long and meaningful captions) - it is often very difficult to read long captions that have very long line lengths, so constraining to the body is actually a feature in these cases. I think these are likely the most common cases (at least in our current thinking), so may want to consider some way to specifically set those caption locations / columns.

**Comment by batpigandme** [comment]
Point well made (and taken), Charles. Realistically, I can't imagine wanting the caption to be longer than the body column, and I hadn't considered what it would look like for a caption with long text. I tested (and added to the example repo/page) a case where I'd wan't the `<figcaption>` to follow the image layout (I had two grid columns inside of a `.column-page`), and those work just fine (since the `column-page` div isn't the direct parent of the images).
![Screenshot of page with column-page div highlighted that contains two images inside of bootstrap grid columns.](https://github.com/quarto-dev/quarto-cli/assets/831732/119cd316-b08f-4435-97c4-c3d1369cdb71)
Given that, I imagine it wouldn't be that hard to hack around the default figcaption layout as is. So, I think it's safely low priority! 😊

**Comment by OrenBochman** [comment]
It seems like a high impact issue please give it more immediate attention.
 


Hi, I have a similar issue with caption of basic image in HTML.

Almost 95% of images in my quarto are tucked in the margin and make use of expansion via a lightbox filter.

```
![a caption](file_name.png){.column-margin}
```

**Since upgrading to 1.4, the caption for these appear in the body column while, the image is now the margin.**

Right now, my workaround is to fencing new images with a div, ie:

```
::: column-margin
![a caption](file_name.png){.column-margin}
:::
```
however, there is no simple way to automate this as existing images may be in divs or in <p> elements. and some images have extra attributes in the {} component.

This is a big headache - since there are thousands of links like this.... e.g. screen caps from many lectures. The web standards mandates using captions for SEO but also to support screen readers for people with disablilites.

I suggest updating the priority as this is impacting anyone using a captioned image as part of the marginellia and is breaking layout after upgrading to version 1.4.

Note: I recently reviewed 5+ blog post by users instructing on migration of blogs etc. to qurto and pretty much everyone  is using the margin one way or another.

**Comment by batpigandme** [comment]
Hey @OrenBochman,

I just added an example with the basic syntax (i.e. no special surrounding div) to the demo page I made, and it's possible this has already been fixed in the development version. Do you know if this _only_ happens with the lightbox feature? (I don't have that turned on…not gonna lie, I've actually never used it).
<img width="1359" alt="Screenshot of demo page with figure in the margin column. The caption appears to go under the margin figure." src="https://github.com/quarto-dev/quarto-cli/assets/831732/ab153743-9a80-4552-8e1e-12629d733ef7">

**Comment by cscheid** [comment]
> It seems like a high impact issue please give it more immediate attention.

@OrenBochman I just want to make it clear that we decide what to work on based on what we perceive is most important. We were working on it over the _weekend_. Please be more considerate in your wording in the future.

**Comment by dragonstyle** [comment]
I was able to reproduce this when using lightbox + margin captions. I'm going to go ahead and open a separate issue to track this as it isn't the same issue as this.

**Comment by dragonstyle** [comment]
https://github.com/quarto-dev/quarto-cli/issues/8697

**Comment by OrenBochman** [comment]
> @OrenBochman I just want to make it clear that we decide what to work on based on what we perceive is most important. We were working on it over the _weekend_. Please be more considerate in your wording in the future.

I'd like to apologize to everyone, as my note was out of line.  
I will try to do better going forward. 

Also, I understand the code in the screenshot better and realize that this is a more nuanced issue and not the bug of the basic :lady_beetle:  I thought it was. 

I salute :vulcan_salute: the work you are doing on Quarto, thanks for the assistance!

**Comment by OrenBochman** [comment]
> Hey @OrenBochman,
> 
> I just added an example with the basic syntax (i.e. no special surrounding div) to the demo page I made ([source code](https://github.com/batpigandme/quarto-figcap-demo/blob/dd553c2f501d183959f028583c4d7e8db5fd44f3/index.qmd#L78), [live demo](https://mara.quarto.pub/quarto-figcap-demo/#margin-figures-and-images)), and it's possible this has already been fixed in the development version (which is what I use on my machine). 
> 
> Do you know if this _only_ happens with the lightbox feature? 
> <img width="1359" alt="Screenshot of demo page with figure in the margin column. The caption appears to go under the margin figure." src="https://github.com/quarto-dev/quarto-cli/assets/831732/ab153743-9a80-4552-8e1e-12629d733ef7">
> 
> P.S. Struck my comment re. priority—obviously I haven't gone through all the use cases, and thought it was kind of a niche thing, since I hadn't seen anyone else post about it. 🙂 
> 
> **Update** Added an example that uses `.lightbox` to the demo, and the figcaption location seems fine (though the layout of the figures is overlapping in the version on quarto-pubs, but not locally—I think that's unrelated).

 last I checked @dragonstyle  opened a new bug and fixed it (by promoting the collum-margin css selector to the figure.) I wouldn't mind using a nightly build as I have couple of fixed issues that are bothersome to work with.

**Comment by dragonstyle** [comment]
We have fresh pre-release builds that include the fix (any of the newer 1.5 builds will have it- there is a fresh one as of this morning). 

https://quarto.org/docs/download/

It will also be in our next 1.4 patch release, which is imminent, but not 100% sure when that will arrive.
